components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt

```diff
@@ -48,6 +49,21 @@ class Session(
      */
     data class SecurityInfo(val secure: Boolean = false, val host: String = "", val issuer: String = "")
 
+    /**
+     * Represents the origin of a session to describe how and why it was created.
+     */
+    enum class Source {
+        ACTION_SEND, // Created to handle an ACTION_SEND (share) intent
+        ACTION_VIEW, // Created to handle an ACTION_VIEW intent
+        CUSTOM_TAB, // Created to handle a CustomTabs intent
+        HOME_SCREEN, // User interacted with the home screen
+        MENU, // User interacted with a menu
+        NEW_TAB, // User opened a new tab
+        NONE, // Default value and for testing purposes
+        TEXT_SELECTION, // Default value and for testing purposes
+        USER_ENTERED //  User entered a URL or search term
```

Comment:
Why not a JavaDoc (or however it's called for Kotlin code 😀) so that it shows up in the IDE when using the enum?